prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -87,18 +199,23 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
          * @return the timestamp or null if not specified
          */
         fun getMessageTimestamp(message: Message): Date? {
-            val timestamp = (message["MSH"] as MSH).msh7_DateTimeOfMessage
-            return if (!timestamp.isEmpty && !timestamp.ts1_Time.isEmpty) {
-                timestamp.ts1_Time.valueAsDate
-            } else null
+            return when (val structure = message["MSH"]) {
```

Comment:
NIT: Where we use these strings, can we get this value from some enum in an existing library instead?

Reply:
Didn't see any enum values for this so I just pulled this up as a constant